File: server/router/business/basic_file.go
package business

import (
	"go-cms/api/v1"
	"go-cms/middleware"
	"github.com/gin-gonic/gin"
)

type BasicFileRouter struct {
}

// InitBasicFileRouter 初始化 BasicFile 路由信息
func (s *BasicFileRouter) InitBasicFileRouter(Router *gin.RouterGroup) {
	basicFileRouter := Router.Group("basicFile").Use(middleware.OperationRecord())
	basicFileRouterWithoutRecord := Router.Group("basicFile")
	var basicFileApi = v1.ApiGroupApp.BusinessApiGroup.BasicFileApi
	{
		basicFileRouter.POST("createBasicFile", basicFileApi.CreateBasicFile)   // 新建BasicFile
		basicFileRouter.DELETE("deleteBasicFile", basicFileApi.DeleteBasicFile) // 删除BasicFile
		basicFileRouter.DELETE("deleteBasicFileByIds", basicFileApi.DeleteBasicFileByIds) // 批量删除BasicFile
		basicFileRouter.PUT("updateBasicFile", basicFileApi.UpdateBasicFile)    // 更新BasicFile
	    basicFileRouter.POST("quickEdit", basicFileApi.QuickEdit)  // 快速编辑
		basicFileRouterWithoutRecord.GET("excelList", basicFileApi.ExcelList)  // 分页导出excel BasicFile列表
	}
	{
		basicFileRouterWithoutRecord.GET("findBasicFile", basicFileApi.FindBasicFile)        // 根据ID获取BasicFile
		basicFileRouterWithoutRecord.GET("getBasicFileList", basicFileApi.GetBasicFileList)  // 获取BasicFile列表
	    
	}
}

File: web/src/api/basicFile.ts
import { http,resData } from "@/utils/http";
 
// @Tags BasicFile
// @Summary 创建BasicFile
export const createBasicFile = (data?: object) => {
   return http.request<resData>("post", "/basicFile/createBasicFile", { data });  
}

// @Tags BasicFile
// @Summary 删除BasicFile
export const deleteBasicFile = (data?: object) => {
   return http.request<resData>("post", "/basicFile/deleteBasicFile", { data });  
}
 

// @Tags BasicFile
// @Summary 删除ByIdsBasicFile
export const deleteBasicFileByIds = (data?: object) => {
   return http.request<resData>("post", "/basicFile/deleteBasicFileByIds", { data });  
}
 

// @Tags BasicFile
// @Summary 更新BasicFile
export const updateBasicFile = (data?: object) => {
   return http.request<resData>("post", "/basicFile/updateBasicFile", { data });  
}

// @Tags BasicFile
// @Summary 快速编辑
export const quickEdit = (data?: object) => {
   return http.request<resData>("post", "/basicFile/quickEdit", { data });  
}
 

// @Tags BasicFile
// @Summary 用id查询BasicFile
export const findBasicFile = (params?: object) => {
   return http.request<resData>("get", "/basicFile/findBasicFile", { params });  
}
 

// @Tags BasicFile
// @Summary 分页获取BasicFile列表
export const getBasicFileList = (params?: object) => {
   return http.request<resData>("get", "/basicFile/getBasicFileList", { params });  
}
 

// @Tags BasicFile
// @Summary 导出excelBasicFile列表
export const excelList = (params?: object) => {
   return http.request<resData>("get", "/basicFile/excelList", { params });  
}
